lib/datastore/request.js

```diff
@@ -617,11 +616,15 @@ DatastoreRequest.prototype.runQuery = function(query, callback) {
       return;
     }
 
-    var entities = entity.formatArray(resp.batch.entity_result);
+    var entities = [];
     var nextQuery = null;
 
-    if (resp.batch.end_cursor && entities.length > 0) {
-      var endCursor = resp.batch.end_cursor.toBase64();
+    if (resp.batch.entityResults) {
+      entities = entity.formatArray(resp.batch.entityResults);
+    }
+
+    if (resp.batch.endCursor && entities.length > 0) {
```

Comment:
You should use more_results to determine if you need to issue another query (in particular, if it is == NOT_FINISHED.